app/models/vouchers/percentage_rate.rb

```diff
@@ -13,8 +13,11 @@ def display_value
     # We limit adjustment to the maximum amount needed to cover the order, ie if the voucher
     # covers more than the order.total we only need to create an adjustment covering the order.total
     def compute_amount(order)
-      percentage = amount / 100
-      -percentage * order.pre_discount_total
+      rate(order) * order.pre_discount_total
+    end
```

Comment:
I wonder if we should add `compute_tax_amount` to the model as well... or take this out of the model. But then this is to comply with the originator interface of the Spree adjustments, right? Nevermind...

Reply:
Yes the `compute_amount` is there to comply with the originator interface, see `CalculatedAdjustments` concern. That said, I believe voucher adjustments are decoupled from other adjustment lifecycle, so I think we "should be able" to take it out of the model.